ui/app/adapters/kv/metadata.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import ApplicationAdapter from '../application';
+import { kvMetadataPath } from 'vault/utils/kv-path';
+
+export default class KvMetadataAdapter extends ApplicationAdapter {
+  namespace = 'v1';
+
+  _url(fullPath) {
+    return `${this.buildURL()}/${fullPath}`;
+  }
+
+  createRecord(store, type, snapshot) {
+    const { backend, path } = snapshot.record;
+    const id = kvMetadataPath(backend, path);
+    const url = this._url(id);
+    const data = this.serialize(snapshot);
+    return this.ajax(url, 'POST', { data }).then(() => {
+      return {
+        id,
+        data,
+      };
+    });
+  }
+
+  updateRecord(store, type, snapshot) {
+    const { backend, path } = snapshot.record;
+    const id = kvMetadataPath(backend, path);
+    const url = this._url(id);
+    const data = this.serialize(snapshot);
+    return this.ajax(url, 'POST', { data }).then(() => {
+      return {
+        id,
+        data,
+      };
+    });
+  }
+
+  query(store, type, query) {
+    const { backend, pathToSecret } = query;
```

Comment:
I wonder if we should default `pathToSecret` to empty string so it never sets `undefined` on the path

```suggestion
    const { backend, pathToSecret = '' } = query;
```

Reply:
We set it to an empty string in the [model ](https://github.com/hashicorp/vault/pull/22426/files#diff-75318e5b1d561b05685fda550e805ef9c6fd7005d8cba6b06951e926a40b0177R50)as a default before we fire of the query. Do you think that's sufficient. 
`const pathToSecret = path_to_secret ? normalizePath(path_to_secret) : '';`

Reply:
It feels like setting the default for the query arg should happen here in the model instead of the model hook? That's where I would have thought to look for it 💭